src/scripts.c

```diff
@@ -33,11 +33,29 @@
 #include "config.h"
 #include "swupd.h"
 
+static bool in_container(void)
```

Comment:
I would just have changed the
"%s/usr/bin/clr-boot-manager update --path %s"

in update_boot to

"/usr/bin/systemd-detect-virt -c || %s/usr/bin/clr-boot-manager update --path %s"

rather than write this new function.

Reply:
Why? The new function makes it easier to see what is happening and why we return early.

Reply:
1. Easier to see why it is happening is subjective. If you are looking at the output of strace to try and figure out why something is not updating, then seeing `systemd-detect-virt || clr-boot-manager` is a lot easier to reason about than the typical hundreds of lines that process invocation causes.

1. Changing the string is a smaller change.

1. Eventually when it is decided that this kind of policy decision is moved outside the swupd-client to a supporting program (There is already a good argument to be made that this kind of decision is non of swupd's business) then we will only need to change the code in one place.

Random thoughts on what might be needed in the future. A container might be being used to provision a new disk, and so have access to both the EFI and root partitions. With this change we can no longer use swupd to do so, we have to run a clr-boot-manager update as another step.

 Maybe in the future something like clear containers will respond 'true' to being run in a container but still want to be able to update its kernel.

The change is fine from the point of view of code, it is a small function (Although I would have made it a lot smaller by not copying a constant string to a malloced one, using the malloced one, and then freeing it), well named etc.

Reply:
I'd expect the provision to a new disk to use a different root so this change should still allow swupd to call clr-boot-manager in that case.

I just took container detection as something that may move off to a utility at some point in time so I split it into a function. Dropping the string allocation is definitely the right move though, I had briefly considered using the path prefix'd version before I came to my senses =P.